zenodo_accessrequests/views/requests.py

```diff
@@ -156,7 +156,22 @@ def confirm(pid, record, template, **kwargs):
 
     # Confirm email address.
     if r.status != RequestStatus.EMAIL_VALIDATION:
-        abort(404)
+        if(str(r.status.value).lower() == "pending"):
+            abort(
+                400, 
+                (
+                    "Your email address has been verified.The access request "
+                    "is now pending a decision from the record owner."
+                )
+            )
+        else:
+            abort(
+                400, 
+                (
+                    "The request has already been {}."
+                    " Check your mail inbox for the record owner's response."
```

Comment:
```suggestion
                    "The request has already been {}. "
                    "Check your mail inbox for the record owner's response."
```

I have a slight preference for spaces to be at the end of the line for multiline strings.